### ISSUE TITLE
Semiring Either instance is dodgy

### ISSUE THREAD

**Issue by hdgarrood** [purescript/purescript-either#34]
A `Semiring` is supposed to form a commutative monoid under addition, but:

```purescript
> Left "hello" + Left " world" :: Either String Int
(Left "hello")

> Left " world" + Left "hello" :: Either String Int
(Left " world")
```

**Comment by paf31** [comment]
Yes, we should remove it. Maybe we can just add a `Warn` constraint for now.

**Comment by garyb** [comment]
Resolved (kinda) in the 0.12 branch by removing it.
